### PR TITLE
QA/ Fix add component

### PR DESCRIPTION
**Pull request by ronronscelestes**
## What

Made "add component" box in CM clickable

## Demo

Go to a CT in CM
Create or edit an entry
When no entry in rep or not rep component click on the box to create one

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js [hunk 1]
```diff
@@ -30,18 +34,26 @@ const IconButton = styled(BaseButton)`
...
 
   return (
-    <Box
+    <CursorBox
       background="neutral100"
       hasRadius
       borderColor="neutral200"
       paddingTop={9}
       paddingBottom={9}
+      onClick={handleBoxClick}
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js:52]: You don't need any ref here. You can use `onClick={onClick}`

File: packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js [hunk 2]
```diff
@@ -30,18 +34,26 @@ const IconButton = styled(BaseButton)`
...
+
+  const handleBoxClick = () => {
+    if (addEntryButtonRef.current) {
+      addEntryButtonRef.current.click();
+    }
+  };
 
   return (
-    <Box
+    <CursorBox
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js:46]: You need to add `as` and `type` here. We should not use onClick on div.

```js
...
as='button'
type='button'
...
```

File: packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js [hunk 3]
```diff
@@ -30,18 +34,26 @@ const IconButton = styled(BaseButton)`
 
 const ComponentInitializer = ({ isReadOnly, onClick }) => {
   const { formatMessage } = useIntl();
+  const addEntryButtonRef = useRef();
+
+  const handleBoxClick = () => {
+    if (addEntryButtonRef.current) {
+      addEntryButtonRef.current.click();
+    }
+  };
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js:43]: You don't need that

File: packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js [hunk 4]
```diff
@@ -11,6 +11,10 @@ import { Text } from '@strapi/design-system/Text';
 import { pxToRem } from '@strapi/helper-plugin';
 import { getTrad } from '../../utils';
 
+const CursorBox = styled(Box)`
+  cursor: pointer;
+`;
+
 const IconButton = styled(BaseButton)`
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js:18]: Can you change this to a span please.

```js
const IconWrapper = styled.span`...`;
```

File: packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js (hunk 2 above)
Comment (HichamELBSI) [reply packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js:46]: and also, add the disabled props `disabled={isReadOnly}`

File: packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js [hunk 5]
```diff
@@ -30,18 +34,26 @@ const IconButton = styled(BaseButton)`
...
       hasRadius
       borderColor="neutral200"
       paddingTop={9}
       paddingBottom={9}
+      onClick={handleBoxClick}
     >
       <Stack size={2}>
         <Flex justifyContent="center" style={{ cursor: isReadOnly ? 'not-allowed' : 'inherit' }}>
-          <IconButton disabled={isReadOnly} onClick={onClick}>
+          <IconButton ref={addEntryButtonRef} disabled={isReadOnly} onClick={onClick}>
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/content-manager/components/ComponentInitializer/index.js:56]: Here, you should use only icon wrapper and remove all the props.

```js
<IconWrapper>
   <PlusCircle />
</IconWrapper>
```

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

**Review by soupette — APPROVED** [review]
LGTM

**Pull request merged**